pkg/hubble/parser/threefour/parser.go

```diff
@@ -147,13 +147,22 @@ func (p *Parser) Decode(data []byte, decoded *pb.Flow) error {
 	if len(data) < packetOffset {
 		return fmt.Errorf("not enough bytes to decode %d", data)
 	}
+	data = data[packetOffset:]
 
 	p.packet.Lock()
 	defer p.packet.Unlock()
 
-	err := p.packet.decLayer.DecodeLayers(data[packetOffset:], &p.packet.Layers)
-	if err != nil {
-		return err
+	// Since v1.1.18, DecodeLayers returns a non-nil error for an empty packet, see
+	// https://github.com/google/gopacket/issues/846
+	// TODO: reconsider this check if the issue is fixed upstream
+	if len(data) > 0 {
```

Comment:
We already check for data len on line 147, maybe we should combine somehow? Aside from breaking tests (are they even properly set up?) what does it mean to not have any layers in a packet?

Reply:
To me it looks like we want to error out at line 147 because the packet is too short to contain a monitor header (for lack of a better term), but the `len(data) == 0` check applies to the data after the monitor header, i.e. there is some info in the monitor header which we potentially want to report. I agree that re-slicing `data` makes that logic harder to follow, so checking `len(data[packetOffset:) == 0` is probably better here.

Reply:
Re. breaking tests: To be honest, I don't know whether this can happen aside from tests, i.e. with real monitor traffic. I only noticed when bumping `gopacket` that said tests are failing. To me it looks like the tests could also be extended to contain some packet data. But then again, given that gopacket accepted empty packets just fine in previous versions (and will do so again if the proposed upstream fix is applied), I figured we rather follow that behavior.